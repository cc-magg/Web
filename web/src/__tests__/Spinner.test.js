import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SpinnerContainer from '../components/spinner/container'
import SpinnerComponent from '../components/spinner/component'

test('Spinner should render as expected', () => {
    const wrapper = shallow(<SpinnerContainer />);
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
});

test('Spinner component should recive the default props', () => {
    const wrapper = shallow(<SpinnerContainer />);
    const child = wrapper.find(SpinnerComponent);
    
    expect(child.props().type).toEqual("spinner-border");
    expect(child.props().color).toEqual("text-primary");
});

test('SpinnerContainer should render as expected', () => {
    const wrapper = shallow(<SpinnerComponent />);
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
});
