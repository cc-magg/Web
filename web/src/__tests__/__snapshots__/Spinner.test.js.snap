// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render as expected 1`] = `
<_default
  align=""
  color="text-primary"
  margin=""
  padding=""
  type="spinner-border"
/>
`;

exports[`SpinnerContainer should render as expected 1`] = `
<div
  className="jsx-3507787210 undefined"
>
  <div
    className="jsx-3507787210 undefined undefined undefined style"
    role="status"
  >
    <span
      className="jsx-3507787210 sr-only"
    >
      Loading...
    </span>
  </div>
  <JSXStyle
    dynamic={
      Array [
        undefined,
      ]
    }
    id="2414451809"
  >
    .style.__jsx-style-dynamic-selector{color:undefined;}
  </JSXStyle>
</div>
`;
